.CMake/compiler-flags.cmake

```diff
@@ -0,0 +1,85 @@
+if(CMAKE_C_COMPILER_ID MATCHES "Clang")
+    add_compile_options(-Werror)
+    add_compile_options(-Wall)
+    add_compile_options(-Wextra)
+    add_compile_options(-Wpedantic)
+    #add_compile_options(-Wreserved-id-macro)
+    #add_compile_options(-Wbad-function-cast)
+    # valid BIKE code sets this off
+    add_compile_options(-Wno-missing-braces)
```

Comment:
For warnings that only affect individual implementations, can we put them in their respective CMakeLists.txt file?